--- a/data/themes/Industrial/greeter-gtk3.ui	2019-02-24 19:36:39.000000000 +0300
+++ b/data/themes/Industrial/greeter-gtk3.ui	2019-11-20 09:29:23.143432758 +0300
@@ -67,7 +67,7 @@
                             <property name="can_focus">False</property>
                             <property name="hscrollbar_policy">GTK_POLICY_NEVER</property>
                             <property name="vscrollbar_policy">GTK_POLICY_NEVER</property>
-                            <property name="min-content-height">64</property>
+                            <property name="min-content-height">0</property>
                             <child>
                               <object class="GtkIconView" id="user_list">
                                 <property name="visible">True</property>
@@ -84,7 +84,7 @@
                           <object class="GtkEntry" id="login_entry">
                             <property name="visible">True</property>
                             <property name="can_focus">True</property>
-                            <property name="invisible_char">&#x2022;</property>
+                            <property name="invisible_char">&#42;</property>
                           </object>
                           <packing>
                             <property name="expand">False</property>
