--- a/htmlreport/cppcheck-htmlreport	2020-05-10 12:31:19.000000000 +0300
+++ b/htmlreport/cppcheck-htmlreport	2020-05-28 16:11:33.140446305 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 from __future__ import unicode_literals
 
--- a/htmlreport/setup.py	2020-05-10 12:31:19.000000000 +0300
+++ b/htmlreport/setup.py	2020-05-28 16:11:42.513153327 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 from setuptools import setup
 
--- a/htmlreport/test_htmlreport.py	2020-05-10 12:31:19.000000000 +0300
+++ b/htmlreport/test_htmlreport.py	2020-05-28 16:11:57.408208939 +0300
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 """Test cppcheck-htmlreport."""
 
 import os
